src/main/java/com/github/princesslana/eriscasper/ErisCasper.java

```diff
@@ -26,41 +30,107 @@
   private final ObjectMapper jackson = Jackson.newObjectMapper();
   private final Payloads payloads = new Payloads(jackson);
 
+  private final Integer[] shard;
+
   private final Routes routes;
 
-  private ErisCasper(BotToken token) {
-    this.token = token;
+  private Gateway gateway;
 
+  private ErisCasper(BotToken token, Integer[] shard) {
+    this.token = token;
+    this.shard = shard;
     routes = new Routes(token, httpClient, jackson);
   }
 
   private Observable<Event> getEvents() {
-    Gateway gateway = Gateway.create(httpClient, payloads);
-
+    gateway = Gateway.create(httpClient, payloads);
     return Single.just(RouteCatalog.getGateway())
         .observeOn(Schedulers.io())
         .flatMap(routes::execute)
         .toObservable()
-        .flatMap(gr -> gateway.connect(gr.getUrl(), token))
+        .flatMap(gr -> gateway.connect(gr.getUrl(), token, shard))
         .observeOn(Schedulers.computation())
         .share();
   }
 
+  public Integer getShardNum() {
+    return shard == null ? -1 : shard[0];
+  }
+
   public void run(Bot bot) {
     Observable<Event> events = getEvents();
 
     RepositoryManager rm = RepositoryManager.create(events);
-
-    bot.apply(new BotContext(events, routes, rm))
+    bot.apply(new BotContext(this, events, routes, rm))
         .doOnError(t -> LOG.warn("Exception thrown by Bot", t))
         .blockingAwait();
   }
 
+  public void shutdownGracefully() {
+    gateway.shutdownGracefully();
+  }
+
+  public Completable requestGuildMembers(RequestGuildMembers request) {
+    return gateway.requestGuildMembers(request);
+  }
+
+  public Completable updatePresence(UpdatePresence update) {
+    return gateway.updatePresence(update);
+  }
+
+  public Completable updateVoiceState(UpdateVoiceState update) {
+    return gateway.updateVoiceState(update);
+  }
+
   public static ErisCasper create() {
-    return create(System.getenv("EC_TOKEN"));
+    String token;
+    if (System.getenv().containsKey("EC_TOKEN")) {
```

Comment:
Using custom and varied handling of system properties/environment variables is just going to be more code we have to change when #106 gets implemented.

I'd very much prefer to use environment variables only until #106 is done.